File: 2026/문제해결기법/2023 중간/01_sequence.c
#include <stdio.h>
#include <stdlib.h>

int is_sequence(int n, int* arr){
	int memory = -1;
	for (int i = 0; i < n-1; i++){

		int GAP = arr[i+1] - arr[i];

		if (memory == -1){
			memory = GAP;
		}
		if (GAP != memory){
			return 0;
		}
	}
	return 1;
}

int main(void){

	int N;
	scanf("%d", &N);
	
	int* ARR = (int*)malloc(sizeof(int) * N);
	for (int i = 0; i < N; i++){
		scanf("%d", &ARR[i]); //C scanf: 공백 알아서 건너뛰고 버퍼 숫자만 찾아옴
	}

	printf("%d", is_sequence(N, ARR));
	
	free(ARR);
}

File: 2026/문제해결기법/2023 중간/01_sequence.py
'''
[ 등차수열 ] 

# input ====
N
(N개만큼 띄어쓰기 간격으로 배열값들 입력받음)

# output ====
(등차수열이면 1 / 아니면 0)

# 접근전략 ====
/ 등차수열 = 각 수열간의 간격(차)이 일정한 수열
-> A[i+1] - A[i] 값이 동일헤야함

'''
import sys
input = sys.stdin.readline

# logic ====
def is_sequence(N):
    global ARR
    
    mem = -1
    for i in range(N-1):
        gap = ARR[i+1] - ARR[i]
        if mem == -1:
            mem = gap
        if gap != mem:
            return 0
    
    return 1
    

# input ====
N = int(input())
ARR = list(map(int, input().split()))

# output ====
print(is_sequence(N))